src/ucp/api/ucp.h

```diff
@@ -186,6 +187,34 @@ enum ucp_dt_type {
 
 /**
  * @ingroup UCP_DATATYPE
+ * @brief Generate an identifier for Scatter-gather IOV data type.
+ *
+ * This macro creates an identifier for datatype of scatter-gather list
+ * with multiple pointers
+ *
+ * @return Data-type identifier.
+ */
+#define ucp_dt_make_iov() (UCP_DATATYPE_IOV)
+
+
+/**
+ * @ingroup UCP_DATATYPE
+ * @brief Structure for scatter-gather I/O.
+ *
+ * Specifies a list of buffers which can be used within a single data transfer
+ * function call.
+ *
+ * @note If @a length is zero, the memory pointed to by @a buffer
+ *       will not be accessed. Otherwise, @a buffer must point to valid memory.
+ */
+typedef struct ucp_dt_iov {
+    void   *buffer;   /**< Data buffer */
```

Comment:
@manjugv @abouteiller @bosilca - guys - can you please take at look at the documentation part.